### ISSUE TITLE
can't install "-pre-1" packages

### ISSUE THREAD

**Issue by bukzor** [Yelp/venv-update#61]
This might be a py35 issue. The crucial bit is `UserWarning: Normalizing '1.4.0-pre-1' to '1.4.0rc1'`.

```
$ cat requirements.txt
docker-py==1.4.0-pre-1

$ PIP_INDEX_URL=https://pypi.yelpcorp.com/simple python ../venv_update.py
Keeping virtualenv from previous run.
> pip --version
pip 7.0.1 from /home/buck/venv/mypy/lib/python3.5/site-packages (python 3.5)
> virtualenv_run/bin/python -m pip.__main__ install 'pip>=1.5.0,<6.0.0'
Requirement already satisfied (use --upgrade to upgrade): pip>=1.5.0,<6.0.0 in ./virtualenv_run/lib/python3.5/site-packages
Cleaning up...
> virtualenv_run/bin/python ../venv_update.py --stage2 virtualenv_run requirements.txt
> pip install --upgrade --use-wheel --download-cache=/home/buck/.pip/cache --find-links=file:///home/buck/.pip/wheelhouse wheel==0.24.0
Requirement already up-to-date: wheel==0.24.0 in ./virtualenv_run/lib/python3.5/site-packages
Cleaning up...
> pip wheel --wheel-dir=/home/buck/.pip/wheelhouse wheel==0.24.0 --download-cache=/home/buck/.pip/cache --find-links=file:///home/buck/.pip/wheelhouse --requirement=requirements.txt
Downloading/unpacking wheel==0.24.0
  File was already downloaded /home/buck/.pip/wheelhouse/wheel-0.24.0-py2.py3-none-any.whl
Downloading/unpacking docker-py==1.4.0-pre-1 (from -r requirements.txt (line 1))
  Using download cache from /home/buck/.pip/cache/https%3A%2F%2Fpypi.yelpcorp.com%2Fpackages%2Fdocker-py-1.4.0-pre-1.tar.gz
  Running setup.py (path:/home/buck/trees/yelp/venv-update/tmp/virtualenv_run/build/docker-py/setup.py) egg_info for package docker-py
    /home/buck/trees/yelp/venv-update/tmp/virtualenv_run/lib/python3.5/site-packages/setuptools/dist.py:285: UserWarning: Normalizing '1.4.0-pre-1' to '1.4.0rc1'
      normalized_version,

    warning: no files found matching 'requirements3.txt'
  Requested docker-py==1.4.0-pre-1 (from -r requirements.txt (line 1)), but installing version 1.4.0rc1
Downloading/unpacking requests>=2.5.2 (from docker-py==1.4.0-pre-1->-r requirements.txt (line 1))
  Downloading requests-2.7.0-py2.py3-none-any.whl (470kB): 470kB downloaded
  Saved /home/buck/.pip/wheelhouse/requests-2.7.0-py2.py3-none-any.whl
  Storing download in cache at /home/buck/.pip/cache/https%3A%2F%2Fpypi.yelpcorp.com%2Fpackages%2Frequests-2.7.0-py2.py3-none-any.whl
Downloading/unpacking six>=1.3.0 (from docker-py==1.4.0-pre-1->-r requirements.txt (line 1))
  File was already downloaded /home/buck/.pip/wheelhouse/six-1.9.0-py2.py3-none-any.whl
Downloading/unpacking websocket-client>=0.32.0 (from docker-py==1.4.0-pre-1->-r requirements.txt (line 1))
  Downloading websocket_client-0.32.0-py3-none-any.whl (196kB): 196kB downloaded
  Saved /home/buck/.pip/wheelhouse/websocket_client-0.32.0-py3-none-any.whl
  Storing download in cache at /home/buck/.pip/cache/https%3A%2F%2Fpypi.yelpcorp.com%2Fpackages%2Fwebsocket_client-0.32.0-py3-none-any.whl
Building wheels for collected packages: docker-py
  Running setup.py bdist_wheel for docker-py
  Destination directory: /home/buck/.pip/wheelhouse
Successfully built docker-py
Cleaning up...
> pip install --upgrade --use-wheel --download-cache=/home/buck/.pip/cache --find-links=file:///home/buck/.pip/wheelhouse --no-index --requirement=requirements.txt
Ignoring indexes: https://pypi.yelpcorp.com/simple
Downloading/unpacking docker-py==1.4.0-pre-1 (from -r requirements.txt (line 1))
  Could not find a version that satisfies the requirement docker-py==1.4.0-pre-1 (from -r requirements.txt (line 1)) (from versions: 1.4.0rc1)
Cleaning up...
No distributions matching the version for docker-py==1.4.0-pre-1 (from -r requirements.txt (line 1))
Storing debug log for failure in /home/buck/.pip/pip.log

Something went wrong! Sending 'virtualenv_run' back in time, so make knows it's invalid.
Waiting for all subprocesses to finish...
DONE
> touch virtualenv_run --reference requirements.txt --date '1 day ago'
```

**Comment by campaul** [comment]
We saw this again, with '-dev0' normalizing to '_dev0'.

**Comment by bukzor** [comment]
Ran into this again. This is actually dependant on the setuptools version.
They added this version-normalizing behavior sometime between 3.4.4 and 18.4.

**Comment by asottile** [comment]
This is fixed by #170 

```
$ python pip_faster.py install -r reqs.txt --prune -i https://pypi.yelpcorp.com/simple
Collecting docker-py==1.4.0-pre-1 (from -r reqs.txt (line 1))
  Downloading https://pypi.yelpcorp.com/api/package/docker-py/docker_py-1.4.0_pre_1-py2-none-any.whl
Collecting requests>=2.5.2 (from docker-py==1.4.0-pre-1->-r reqs.txt (line 1))
  slow: full search for unpinned requirement requests>=2.5.2 (from docker-py==1.4.0-pre-1->-r reqs.txt (line 1))
  Downloading https://pypi.yelpcorp.com/api/package/requests/requests-2.12.4-py2.py3-none-any.whl (576kB)
    100% |████████████████████████████████| 583kB 1.1MB/s 
Collecting websocket-client>=0.32.0 (from docker-py==1.4.0-pre-1->-r reqs.txt (line 1))
  slow: full search for unpinned requirement websocket-client>=0.32.0 (from docker-py==1.4.0-pre-1->-r reqs.txt (line 1))
  Downloading https://pypi.yelpcorp.com/api/package/websocket-client/websocket_client-0.37.0-py2-none-any.whl (197kB)
    100% |████████████████████████████████| 204kB 1.1MB/s 
Collecting six>=1.3.0 (from docker-py==1.4.0-pre-1->-r reqs.txt (line 1))
  slow: full search for unpinned requirement six>=1.3.0 (from docker-py==1.4.0-pre-1->-r reqs.txt (line 1))
  Downloading https://pypi.yelpcorp.com/api/package/six/six-1.10.0-py2.py3-none-any.whl
Requirement already satisfied: argparse in /usr/lib/python2.7 (from websocket-client>=0.32.0->docker-py==1.4.0-pre-1->-r reqs.txt (line 1))
Collecting backports.ssl-match-hostname (from websocket-client>=0.32.0->docker-py==1.4.0-pre-1->-r reqs.txt (line 1))
  slow: full search for unpinned requirement backports.ssl-match-hostname (from websocket-client>=0.32.0->docker-py==1.4.0-pre-1->-r reqs.txt (line 1))
  Downloading https://pypi.yelpcorp.com/api/package/backports-ssl-match-hostname/backports.ssl_match_hostname-3.5.0.1-py2-none-any.whl
Installing collected packages: requests, six, backports.ssl-match-hostname, websocket-client, docker-py
Successfully installed backports.ssl-match-hostname-3.5.0.1 docker-py-1.4.0rc1 requests-2.12.4 six-1.10.0 websocket-client-0.37.0
```
